gradle-baseline-java/src/main/java/com/palantir/baseline/tasks/CheckJUnitDependencies.java

```diff
@@ -49,178 +42,122 @@ public CheckJUnitDependencies() {
 
     @TaskAction
     public final void validateDependencies() {
-        getProbablyTestSourceSets().forEach(ss -> {
-            Optional<Test> maybeTestTask = BaselineTesting.getTestTaskForSourceSet(getProject(), ss);
-            if (!maybeTestTask.isPresent()) {
-                // source set doesn't have a test task, e.g. 'schema'
-                return;
-            }
-            Test task = maybeTestTask.get();
-
-            getProject().getLogger().info("Analyzing source set {} with task {}", ss.getName(), task.getName());
-            validateSourceSet(ss, task);
-        });
-    }
+        getProject()
+                .getExtensions()
+                .getByType(JavaPluginExtension.class)
+                .getSourceSets()
+                .forEach(ss -> {
+                    Test task = getProject().getTasks().withType(Test.class).findByName(ss.getName());
+                    if (task == null) {
+                        return;
+                    }
 
-    @Classpath
-    public final Provider<List<Configuration>> getConfigurations() {
-        return getProject().provider(() -> getProbablyTestSourceSets()
-                .map(SourceSet::getRuntimeClasspathConfigurationName)
-                .map(getProject().getConfigurations()::getByName)
-                .collect(toList()));
-    }
+                    getLogger().info("Analyzing source set {} with task {}", ss.getName(), task.getName());
 
-    private Stream<SourceSet> getProbablyTestSourceSets() {
-        return getProject().getExtensions().getByType(JavaPluginExtension.class).getSourceSets().stream()
-                .filter(ss -> !ss.getName().equals(SourceSet.MAIN_SOURCE_SET_NAME));
```

Comment:
I've removed this filtering because:
- It's inconsistent with the `BaselineTesting` plugin, where we don't filter the source sets.
- It's almost certainly unnecessary. If it's a `Test` task, we want to perform this validation.